azure/scope/managedcontrolplane.go

```diff
@@ -198,6 +198,12 @@ func (s *ManagedControlPlaneScope) NodeSubnet() infrav1.SubnetSpec {
 	}
 }
 
+// SetSubnet sets the passed subnet spec into the scope.
+// This is not used when using a managed control plane.
+func (s *ManagedControlPlaneScope) SetSubnet(subnetSpec infrav1.SubnetSpec) {
+
```

Comment:
```suggestion
    return // no-op
```

Reply:
If I add the `return` I get a linting error
```
S1023: redundant `return` statement (gosimple)
```

Reply:
yeah, I think you can't return because the func has no return type

Reply:
are you okay with keeping it empty @CecileRobertMichon ?

Reply:
yeah that's fine